app/apollo/test/subscriptions.fga.spec.js

```diff
@@ -0,0 +1,659 @@
+        expect( subscription.identitySyncStatus.unknownCount, 'subscription identitySyncStatus.unknownCount should be zero' ).to.equal(0);
+      }
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // subscriptions for fgaUser02
+  it('fgaUser02 has authorization to get ALLOWED subscription 2', async () => {
+    let response;
+    try {
+      response = await subscriptionApi.subscriptions(fgaToken02, {
+        orgId: org01._id,
+      });
+      const subscriptions = response.data.data.subscriptions;
+      expect(subscriptions).to.have.length(1);
+      expect(subscriptions[0].name).to.equal(testSubscription2.name);
+      for( const subscription of subscriptions ) {
+        expect( subscription.identitySyncStatus, 'subscription did not include identitySyncStatus' ).to.exist;
+        expect( subscription.identitySyncStatus.syncedCount, 'subscription identitySyncStatus.syncedCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.failedCount, 'subscription identitySyncStatus.failedCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.pendingCount, 'subscription identitySyncStatus.pendingCount should be zero' ).to.equal(0);
+        expect( subscription.identitySyncStatus.unknownCount, 'subscription identitySyncStatus.unknownCount should be zero' ).to.equal(0);
+      }
+    } catch (error) {
+      console.error(JSON.stringify({'API response:': response && response.data ? response.data : 'unexpected response'}, null, 3));
+      console.error('Test failure, error: ', error);
+      throw error;
+    }
+  });
+
+  // subscription
+  it('fgaUser01 has authorization to get subscription by subscription 1 uuid', async () => {
+    let response;
+    try {
+      response = await subscriptionApi.subscription(fgaToken01, {
+        orgId: org01._id,
+        uuid: testSubscription1.uuid,
+      });
+      expect(response.data.errors).to.be.undefined;
+      const subscription = response.data.data.subscription;
+      expect(subscription.name).to.equal(testSubscription1.name);
+      expect(subscription.groupObjs).to.exist;
+      expect(subscription.groupObjs.length).to.equal(1);
+      expect(subscription.groupObjs[0].uuid).to.equal(testGroup1.uuid);
+      expect(subscription.versionObj).to.exist;
+      expect(subscription.versionObj.name).to.equal( testVersion1.name );
+      expect(subscription.versionObj.remote).to.exist;
+      expect(subscription.versionObj.remote.parameters).to.exist;
+      expect(subscription.versionObj.remote.parameters.length).to.equal(1);
+      expect(subscription.versionObj.remote.parameters[0].key).to.equal( 'key1' );
+      expect(subscription.versionObj.remote.parameters[0].value).to.equal( 'val1' );
```

Comment:
```suggestion
      expect(subscription.versionObj.remote.parameters.length).to.equal( testVersion1.remote.parameters.length );
      expect(subscription.versionObj.remote.parameters[0].key).to.equal( testVersion1.remote.parameters[0].key );
      expect(subscription.versionObj.remote.parameters[0].value).to.equal( testVersion1.remote.parameters[0].value );
```
Avoid using hard-coded values that **_must_** match the defined constants like `testVersion1` object.

Arguably you don't even need to be checking _every single attribute_ of the response either -- this isn't necessary to verify `fgaUser01 has authorization to get subscription by subscription 1 uuid`.  If the response has a subscription and the subscription has the right name, the test is good..... unless you're also trying to verify something else not mentioned in the test description?

Reply:
Checked through fga files and deleted quite a few unneeded expect lines with hardcoded values.

Reply:
Resolved